### PR TITLE
fix(js): inbox notifications component gets remounting when render notification prop changes

### PR DESCRIPTION
**Pull request by LetItRock**
### What changed? Why was the change needed?
Fixed the issue described in this ticket: https://linear.app/novu/issue/COM-199/the-inbox-notifications-component-gets-remounted-when

### Screenshots

https://github.com/user-attachments/assets/316aba7f-48d8-45cd-99fc-17718ced88d8

### CODE REVIEW

File: packages/js/src/ui/components/Renderer.tsx [hunk 1]
```diff
@@ -75,13 +79,14 @@ export const Renderer = (props: RendererProps) => {
           <FocusManagerProvider>
             <InboxProvider tabs={props.tabs}>
               <CountProvider>
-                <For each={[...props.nodes]}>
-                  {([node, component]) => {
+                <For each={nodes()}>
```
Comment (LetItRock) [inline packages/js/src/ui/components/Renderer.tsx:80]: The nodes aka DOM element refs don't change on re-render, so it's "safe" to use them in the `For` component `each` prop as the Solid is doing comparison based on refs when rendering DOM elements. This code will make sure that the corresponding Solid components won't remount on props change.

File: packages/js/src/ui/components/Renderer.tsx [hunk 2]
```diff
@@ -75,13 +79,14 @@ export const Renderer = (props: RendererProps) => {
...
+                  {(node) => {
+                    const novuComponent = createMemo(() => props.nodes.get(node)!);
```
Comment (LetItRock) [inline packages/js/src/ui/components/Renderer.tsx]: However, the Solid component props might change, which is why we should memoize it and update the signal when it has been changed.

File: playground/nextjs/src/pages/notifications/index.tsx [hunk 3]
```diff
@@ -1,16 +1,53 @@
 import Title from '@/components/Title';
 import { novuConfig } from '@/utils/config';
 import { Inbox, Notifications } from '@novu/react';
+import { useState } from 'react';
 
 export default function Home() {
+  const [count, setCount] = useState(0);
```
Comment (LetItRock) [inline playground/nextjs/src/pages/notifications/index.tsx:7]: This is the playground code to verify that the state changes are propagated but the `Notifications` component doesn't get remounted.

**Review by SokratisVidros — APPROVED** [review]
💯

**Review by desiprisg — APPROVED** [review]
Looks good! Let's replace the memo in JSX with just an arrow function!

**Pull request merged**